src/server/master_server/metadata_manager.cc

```diff
@@ -0,0 +1,132 @@
+#include "metadata_manager.h"
+#include <stack>
+
+using protos::FileMetadata;
+
+namespace gfs {
+namespace server {
+
+MetadataManager::MetadataManager() {
+  lockManager_ = LockManager::GetInstance();
+  fileMetadataLock_ = new absl::Mutex();
+}
+
+google::protobuf::util::Status MetadataManager::CreateFileMetadata(
+  const std::string& pathname) {
+  
+  // Step 1. Lock (readerlock underneath) the parent directory first
+  ParentLocksAnchor parentLockAnchor(lockManager_, pathname);
+             
+  if (!parentLockAnchor.ok()) {
+    // If this operation fails, which means some of the parent directory does not
+    // exist, we return false
+    return parentLockAnchor.status();
+  }
+
+  // Step 2. Add a new lock for this new file, and writeLock it
+  auto newLockRes(lockManager_->CreateLock(pathname));
```

Comment:
again. don't use `auto` everywhere and don't use `newLockRes` etc. that's not clear in naming. see recommended code example in comment below